### DISCUSSION TITLE
auto-orient changes file size significantly

### DISCUSSION THREAD

**Discussion by jbeyer05** [ImageMagick/ImageMagick#5378] [Help] [answered]
I am running images through convert, and processing them only with the auto-orient command, as follows:

`convert mobile_1653437164.jpg -auto-orient mobile_1653437164.auto-orient.jpg`

I find that ImageMagick frequently drops around 20% from the file size, e.g. original image is 5 MB and the rotated image is 4MB.
I wouldn't have thought that IM should be doing anything to modify quality, etc.

So what is going on here?  I thought it might be an embedded thumbnail that is being removed.  I used exiftool as follows to extract out a thumbnail that was only 2K.  So that doesn't seem to account for the drop of 1 MB.

`exiftool -v -ThumbnailImage mobile_1653437164.jpg > mobile_1653437164.thumb.jpg`

Thanks in advance!

**Comment by fmw42** [comment, marked as answer]
JPG files are lossy compressed. Each time it is read and writing a round of new compression occurs.  So the quality will change.  IM will try to use the input quality value when recompressing, if it is available. If not, it uses 92.  If your input had a quality of 100, but was not recorded, then IM will compresses it at 92, which lower quality than the 100 input.  In most cases, a read and write will result in more compression and quality loss when the same quality value is used for input and output.

**Reply by jbeyer05** [reply]
Thanks fmw42.  I was aware that JPG files are lossy, and it makes sense that JPG -> raw -> new JPG could change the quality, if the quality isn't known.  But I had assumed that with a rotation, IM could do something more intelligent that didn't go all the way to raw and back.  But thanks for the clarification.

**Reply by jbeyer05** [reply]
I just checked the quality before and after rotation, and see that both are 100, according to identify.  So does a 20% filesize reduction still seem logical if quality is 100 both before and after?

```
[03:52 PM] MacBook-Pro :: ~/Downloads $ identify -format '%Q' mobile_1653437164.jpg
100
[03:53 PM] MacBook-Pro :: ~/Downloads $ identify -format '%Q' mobile_1653437164.auto-orient.jpg
100
```

**Comment by fmw42** [comment]
The image has to be read before the rotation. That decompresses it. I know of no way to rotate even 90 deg increments without decompressing images first. Then once rotated, it has to be re-compressed. Since JPG is lossy compared to PNG or TIFF, it can change file size due to changes in compression

I am not an expert on JPG, so cannot say about 20% change. But different compression tables could be involved as well. So I suspect it is possible with everything going on.  Also just because the Q level is unchanged, does not mean that the actual file size would not change. The decompress and recompress losses information and thus the file size would decrease.

**Comment by snibgo** [comment]
@jbeyer05: IM always decompresses, then does whatever operations you ask for, then re-compresses.

Specialist JPEG programs operate with knowledge of JPEG internals, so can take short cuts. For example, "jpegtran".

**Reply by Jmuccigr** [reply]
Yes, I was going to suggest `jpegtran` if all you're doing is rotating jpeg images. jpegtran will also do lossless cropping and other things.

https://www.gsp.com/cgi-bin/man.cgi?topic=JPEGTRAN
